### PR TITLE
implemented update handler. refactored service return type for save a…

### PR DESCRIPTION
**Pull request by markhaur**
implemented update handler. refactored service return type for save and update method. refactored erroneous test cases

### CODE REVIEW

File: checklist/http.go [hunk 1]
```diff
@@ -149,6 +155,38 @@ func (s *server) handleToggleTask() http.HandlerFunc {
...
+
+		var req request
+		if err := json.NewDecoder(r.Body).Decode(&req); err != nil {
+			writeError(w, ErrInvalidRequestBody{err})
+			return
+		}
+
+		task, err := s.service.Update(r.Context(), todo.Task{ID: id, Name: req.Name, Done: req.Done})
+		if err != nil {
+			writeError(w, err)
```
Comment (jarri-abidi) [inline checklist/http.go:183]: we should return here

Comment (markhaur) [reply checklist/http.go:183]: lemme do it

File: checklist/service.go [hunk 2]
```diff
@@ -70,6 +70,16 @@ func (s *service) Remove(ctx context.Context, id int64) error {
...
-func (s *service) Update(ctx context.Context, task *todo.Task) error {
-	return nil
+func (s *service) Update(ctx context.Context, task todo.Task) (*todo.Task, error) {
+	err := s.repository.Update(ctx, &task)
+	if err == todo.ErrTaskNotFound {
+		err = s.repository.Insert(ctx, &task)
+		if err != nil {
+			return nil, fmt.Errorf("could not create task: %v", err)
+		}
+	}
```
Comment (jarri-abidi) [inline checklist/service.go:81]: would it make sense to have 1 repository method like Upsert? I mean to save multiple db calls

Comment (markhaur) [reply checklist/service.go:81]: Lemme think about it till tomorrow.

**Review by jarri-abidi — CHANGES_REQUESTED** [review]
Need to return 201 if newly created

**Review by jarri-abidi — APPROVED** [review]

**Pull request merged**